android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -752,4 +840,310 @@ public static ErcToken createEthereumErcToken() {
         }
         return tokens;
     }
+
+    public static String getAccountName(AccountInfo[] accounts, String address) {
+        String addressLowerCase = address.toLowerCase(Locale.getDefault());
+        for (AccountInfo account : accounts) {
+            if (account.address.toLowerCase(Locale.getDefault()).equals(addressLowerCase)) {
+                return account.name;
+            }
+        }
+
+        return stripAccountAddress(address);
+    }
+
+    public static void openTransaction(
+            TransactionInfo txInfo, EthJsonRpcController ethJsonRpcController, Activity activity) {
+        if (txInfo == null || txInfo.txHash == null || txInfo.txHash.isEmpty()) {
+            return;
+        }
+        assert ethJsonRpcController != null;
+        ethJsonRpcController.getChainId(chainId -> {
+            ethJsonRpcController.getAllNetworks(networks -> {
+                for (EthereumChain network : networks) {
+                    if (!chainId.equals(network.chainId)) {
+                        continue;
+                    }
+                    String blockExplorerUrl = Arrays.toString(network.blockExplorerUrls);
+                    if (blockExplorerUrl.length() > 2) {
+                        blockExplorerUrl =
+                                blockExplorerUrl.substring(1, blockExplorerUrl.length() - 1)
+                                + "/tx/" + txInfo.txHash;
+
+                        TabUtils.openUrlInNewTab(false, blockExplorerUrl);
+                        TabUtils.bringChromeTabbedActivityToTheTop(activity);
+                        break;
+                    }
+                }
+            });
+        });
+    }
+
+    public static void setUpTransactionList(AccountInfo[] accountInfos,
+            AssetRatioController assetRatioController, EthTxController ethTxController,
+            ErcTokenRegistry ercTokenRegistry, String assetSymbol, String contractAddress,
```

Comment:
`String assetSymbol, String contractAddress, int assetDecimals` - these looks pretty much as a part of `ErcToken`. Maybe replace those parameters with one to reduce the number of arguments?

Reply:
I thought about that as well, but to make it happen I would need to make an extra mojo call to pull all tokens, loop them to find that one and save `ErcToken`. Because the above strings are passed between Activity. Do you know how to pass a complex object like `ErcToken` via Intent from one Activity to another?

Reply:
From what I read, you have to have some kind of serializer for all that. Too complex as to me for the current task

Reply:
>Do you know how to pass a complex object like ErcToken via Intent from one Activity to another?
no, I don't know such ways.

Reply:
but it is possible to combine those fields at `AssetDetailActivity.triggerLayoutInflation`

Reply:
it won't be a real `ErcToken`. I mean not all fields would be filled. And someone accidentally would use non defined fields in the future, like logo for example.

Reply:
yes, makes sense